pkg/util/kubernetes/apiserver/openshift.go

```diff
@@ -0,0 +1,65 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2018 Datadog, Inc.
+
+// +build kubeapiserver
+
+package apiserver
+
+import (
+	"errors"
+
+	osq "github.com/openshift/api/quota/v1"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+var ErrNotOpenShift = errors.New("not an OpenShift cluster")
+
+// IsOpenShift detects available endpoints to determine of OpenShift APIs are available
+func (c *APIClient) IsOpenShift() OpenShiftApiLevel {
+	if c.isOpenShift != OpenShiftUnknown {
+		return c.isOpenShift
+	}
+	err := c.Cl.CoreV1().RESTClient().Get().AbsPath("/apis/quota.openshift.io").Do().Error()
```

Comment:
What about using the `Discovery` api here to see what OpenShift group versions are available?

https://github.com/kubernetes/client-go/blob/master/discovery/discovery_client.go#L135

Reply:
Agreed, has this path changed ? Will it potentially change ?

Reply:
Discovery won't play well with the legacy `/oapi/` endpoint as it's openshift-specific. I went with the same logic for both endpoints to have the same code. Both endpoints are documented in 3.10:

- https://docs.openshift.org/latest/rest_api/apis-quota.openshift.io/v1.ClusterResourceQuota.html
- https://docs.openshift.org/latest/rest_api/oapi/v1.ClusterResourceQuota.html

Redhat being Redhat, these endpoints are here to stay a *long* time. The "old" one is not even listed as deprecated, we could even just use this one unconditionally. I went with the new one when available to be future-proof and make the rbac more idiomatic though.